app/models/agents/website_agent.rb

```diff
@@ -259,13 +260,19 @@ def check_urls(in_url)
       end
     end
 
+    def get_valid_url(url)
+      return (url =~ URI::ABS_URI) ? url : URI.escape(url)
```

Comment:
URI.escape is marked as obsolete, so you should reimplement it yourself, based on the definition of unsafe characters in RFC 3986.